treewide: Initial role PR

haslersn opened the pull request:
* [x] Don't hardcode the python version
* [x] Add `README.md`

haslersn committed:
meta/main.yml

haslersn committed:
Initial, working role

SlothOfAnarchy commented on defaults/main.yml:
```diff
@@ -0,0 +1,36 @@
+    file:
+      class: logging.handlers.RotatingFileHandler
+      formatter: precise
+      filename: "{{ synapse_data_dir }}/homeserver.log"
```
Please place logs in `/var/log/synapse`, not in `/etc` or user-configurable
```suggestion
      filename: "/var/log/synapse/homeserver.log"
```

SlothOfAnarchy commented on defaults/main.yml:
```diff
@@ -0,0 +1,36 @@
+      class: logging.handlers.RotatingFileHandler
+      formatter: precise
+      filename: "{{ synapse_data_dir }}/homeserver.log"
+      maxBytes: 104857600
```
```suggestion
      maxBytes: 104857600 # 100MiB
```

SlothOfAnarchy commented on defaults/main.yml:
```diff
@@ -0,0 +1,36 @@
+      backupCount: 10
+      filters: [context]
+      encoding: utf8
+    console:
```
Will this log to systemd's journal?

haslersn replied:
Yes

SlothOfAnarchy commented on defaults/main.yml:
```diff
@@ -0,0 +1,36 @@
+      level: INFO
+  root:
+    level: INFO
+    handlers: [file, console]
```
```suggestion
    handlers:
      - file
      - console
```

SlothOfAnarchy commented on defaults/main.yml:
```diff
@@ -0,0 +1,36 @@
+      filename: "{{ synapse_data_dir }}/homeserver.log"
+      maxBytes: 104857600
+      backupCount: 10
+      filters: [context]
```
```suggestion
      filters:
        - context
```

SlothOfAnarchy commented on defaults/main.yml:
```diff
@@ -0,0 +1,36 @@
+    console:
+      class: logging.StreamHandler
+      formatter: precise
+      filters: [context]
```
```suggestion
      filters:
        - context
```

SlothOfAnarchy commented on tasks/main.yml:
```diff
@@ -0,0 +1,70 @@
+    state: directory
+    owner: "{{ synapse_user }}"
+  loop:
+    - "{{ synapse_data_dir }}"
```
```suggestion
    - "{{ synapse_data_dir }}"
    - /var/log/synapse
```

SlothOfAnarchy commented on tasks/main.yml:
```diff
@@ -0,0 +1,70 @@
+- name: Install Synapse REST Password provider
+  get_url:
+    url: https://raw.githubusercontent.com/kamax-matrix/matrix-synapse-rest-auth/master/rest_auth_provider.py
+    dest: "{{ synapse_install_dir }}/env/lib/python3.5/site-packages/rest_auth_provider.py"
```
Please don't hardcode the python version

haslersn replied:
We could read the site-packages from the virtualenv via `{{ synapse_install_dir }}/env/bin/python -m site`, but the output still has to be parsed.

But how does this work together with upgrading the python version? As it turns out, the ansible pip module (with virtualenv parameter) will not upgrade the python version of an existing virtualenv. For a freshly created virtualenv, by the way, it uses (by default) the python version which ansible has been run with (on the remote). This is in our setup the (on the remote) system-wide installed python3 version.

Of course I think that whether or not the virtualenv is freshly created shouldn't make a difference, for the sake of purity. Therefore we first need to find a solution on how to upgrade the virtualenv, as soon as a new python3 version is deployed system-wide. This solution should of course **not** encompass deleting and recreating the virtualenv on every deployment.

I think that we should solve that problem first. Its solution might change the possibilities how the problem with the hardcoded python version can be solved.

haslersn replied:
Or we could just omit the virtualenv...

haslersn replied:
This is now resolved. See my newest commit on this branch: 1a59cf84dcfaf84935d23f9d8bdc8106731373f0

haslersn replied:
I cannot guarantee that the downloaded `matrix-synapse-rest-auth` plugin will work with other Python versions, though. Do you approve that this is resolved, nonetheless?

haslersn committed:
place logs in /var/log/synapse, not in /etc or user-configurable

Co-Authored-By: Michel Weitbrecht <michel.weitbrecht@stuvus.uni-stuttgart.de>

haslersn committed:
Comment 100MiB as suggested

Co-Authored-By: Michel Weitbrecht <michel.weitbrecht@stuvus.uni-stuttgart.de>

haslersn committed:
YAML-style lists

Co-Authored-By: Michel Weitbrecht <michel.weitbrecht@stuvus.uni-stuttgart.de>

haslersn committed:
YAML-style lists

Co-Authored-By: Michel Weitbrecht <michel.weitbrecht@stuvus.uni-stuttgart.de>

haslersn committed:
YAML-style lists

Co-Authored-By: Michel Weitbrecht <michel.weitbrecht@stuvus.uni-stuttgart.de>

haslersn committed:
Also create /var/log/synapse, as it's needed to store logs

Co-Authored-By: Michel Weitbrecht <michel.weitbrecht@stuvus.uni-stuttgart.de>

haslersn committed:
Create virtualenv per python version

If the system's python version changes, then the path to the virtualenv
changes and hence it's newly created, using the correct python version.

haslersn committed:
defaults: Rename synapse_default_log_config -> synapse_log_config

haslersn committed:
tasks/main.yml: Fix owners, groups and permissions

haslersn committed:
READme.md: init

SlothOfAnarchy commented on README.md:
```diff
@@ -0,0 +1,51 @@
+| `synapse_homeserver_config` |                         | [#configuration](#configuration) |
+| `synapse_install_dir`       | `/opt/synapse`          | Directory to store the synapse installation    |
+| `synapse_log_config`        | See `defaults/main.yml` | Content of `<server_name>.log.config` |
+| `synapse_signing_key`       |                         | [#configuration](#configuration) |
```
Please indent

SlothOfAnarchy commented on defaults/main.yml:
```diff
@@ -0,0 +1,40 @@
+---
+synapse_user: synapse
+synapse_install_dir: /opt/synapse
+synapse_data_dir: /etc/synapse
```
`/etc` is for configuration files, see https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard#Directory_structure

SlothOfAnarchy commented on templates/synapse.service.j2:
```diff
@@ -0,0 +1,17 @@
+Type=notify
+NotifyAccess=main
+User={{ synapse_user }}
+ExecStart={{ synapse_fact_virtualenv_dir }}/bin/python -m synapse.app.homeserver --config-path={{ synapse_data_dir }}/homeserver.yaml
```
not `python3`?

haslersn replied:
No, there's only one python version in a virtualenv.
